egg/core/interaction.py

```diff
@@ -137,10 +105,35 @@ def _check_cat(lst):
                     "Appending empty and non-empty interactions logs. "
                     "Normally this shouldn't happen!"
                 )
-            return torch.cat(lst, dim=0)
+            # is all elem are tensor stack
+            if all([isinstance(x, torch.Tensor) for x in lst]):
+                return torch.stack(lst, dim=0)
```

Comment:
If tensor are of different sizes this would fail. That does not happen with cat. I would stick with cat